### PR TITLE
Initial Contributing and issue/pr templates

### PR DESCRIPTION
**Pull request by stuartcampbell**
These are my first stab at coming up with a `CONTRIBUTING.md` and templates for Pull Request and Isues. 

They are far from perfect, but I think that they are good enough to start the discussion.

### CODE REVIEW

**Comment by stuartcampbell** [comment]
And I have just realised that I haven't actually followed the contributing guide that I just wrote with my commit messages! 🤦‍♂️

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Standard acronyms to start the commit message with are:
+```
+API: an (incompatible) API change
+BLD: change related to building numpy
+BUG: bug fix
+DEP: deprecate something, or remove a deprecated object
+DEV: development tool or utility
+DOC: documentation
+ENH: enhancement
+MAINT: maintenance commit (refactoring, typos, etc.)
```
Comment (tacaswell) [inline .github/CONTRIBUTING.md]: We have been using `MNT` for this.  It is not special enough to get 2 extra letters.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+REV: revert an earlier commit
+STY: style fix (whitespace, PEP8)
+TST: addition or modification of tests
```
Comment (tacaswell) [inline .github/CONTRIBUTING.md]: We should also add `CI` for commits devoted to managing the CI systems.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<!--- Provide a general summary of your changes in the Title above -->
```
Comment (tacaswell) [inline .github/PULL_REQUEST_TEMPLATE.md]: Tad worried about this, a majority of the PRs will be opened by us and this may just get annoyingly verbose.

Comment (stuartcampbell) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe I was just looking to the future! 😄 What about just the stuff about the checkmarks ?

Comment (stuartcampbell) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'll trim it down a bit.

**Comment by danielballan** [comment]
I think this is worth trying as is. Let's see how we like it.

**Comment by prjemian** [comment]
In [synApps motor ](https://github.com/epics-modules/motor) module, we added these two items to the module's ``README.md`` page that help support the type of issue and feature collaboration this issue describes:

    [Report an issue with Motor](https://github.com/epics-modules/motor/issues/new?title=%20ISSUE%20NAME%20HERE&body=**Describe%20the%20issue**%0A%0A**Steps%20to%20reproduce**%0A1.%20Step%20one%0A2.%20Step%20two%0A3.%20Step%20three%0A%0A**Expected%20behaivour**%0A%0A**Actual%20behaviour**%0A%0A**Build%20Environment**%0AArchitecture:%0AEpics%20Base%20Version:%0ADependent%20Module%20Versions:&labels=bug)  
    [Request a feature](https://github.com/epics-modules/motor/issues/new?title=%20FEATURE%20SHORT%20DESCRIPTION&body=**Feature Long Description**%0A%0A**Why%20should%20this%20be%20added?**%0A&labels=enhancement)

**Comment by prjemian** [comment]
Here's what a BlueSky issue request button would look like:

[Report an issue with bluesky](https://github.com/NSLS-II/bluesky/issues/new?title=Provide%20a%20general%20summary%20of%20the%20issue&body=%23%23%20Expected%20Behavior%0A%0A%0A%0A%23%23%20Current%20Behavior%0A%0A%0A%0A%23%23%20Possible%20Solution%0A%0A%0A%0A%23%23%20Steps%20to%20Reproduce%20(for%20bugs)%0A%0A%0A1.%0A2.%0A3.%0A%0A%23%23%20Context%0A%0A%0A%0A%23%23%20Your%20Environment%0A%0A)

**Comment by danielballan** [comment]
That seems useful to me. I opened a PR here: https://github.com/NSLS-II/bluesky/pull/651

**Pull request merged**